File: src/main/java/seremis/geninfusion/soul/allele/AlleleDouble.java
package seremis.geninfusion.soul.allele;

import net.minecraft.nbt.NBTTagCompound;
import seremis.geninfusion.api.soul.EnumAlleleType;
import seremis.geninfusion.soul.Allele;

public class AlleleDouble extends Allele {

    public double value;

    public AlleleDouble(boolean isDominant, double value) {
        super(isDominant, EnumAlleleType.DOUBLE);
        this.value = value;
    }

    public AlleleDouble(Object... args) {
        super(args);
        value = (Double) args[1];
    }

    public AlleleDouble(NBTTagCompound compound) {
        super(compound);
        readFromNBT(compound);
    }

    @Override
    public void writeToNBT(NBTTagCompound compound) {
        super.writeToNBT(compound);
        compound.setDouble("value", value);
    }

    @Override
    public void readFromNBT(NBTTagCompound compound) {
        super.readFromNBT(compound);
        value = compound.getDouble("value");
    }

}

File: src/main/java/seremis/geninfusion/soul/allele/AlleleDouble.scala
package seremis.geninfusion.soul.allele

import net.minecraft.nbt.NBTTagCompound
import seremis.geninfusion.api.soul.EnumAlleleType
import seremis.geninfusion.soul.Allele

class AlleleDouble(isDominant: Boolean, var value: Double) extends Allele(isDominant, EnumAlleleType.DOUBLE) {

    def this(args: AnyRef*) {
        this(args(0).asInstanceOf[Boolean], args(1).asInstanceOf[Double])
    }

    def this(compound: NBTTagCompound) {
        this(false, 0.0F)
        readFromNBT(compound)
    }

    override def writeToNBT(compound: NBTTagCompound) {
        super.writeToNBT(compound)
        compound.setDouble("value", value)
    }

    override def readFromNBT(compound: NBTTagCompound) {
        super.readFromNBT(compound)
        value = compound.getDouble("value")
    }
}
